src/Symfony/Component/Mime/NamedAddress.php

```diff
@@ -24,7 +24,7 @@ public function __construct(string $address, string $name)
     {
         parent::__construct($address);
 
-        $this->name = $name;
+        $this->name = trim(str_replace(["\n", "\r"], '', $name));
```

Comment:
Is it allowed to have `\n` or `\r` in a name? If not, what about using `trim(preg_replace("{[\r\n]+}", '', $name));` instead?

Reply:
Another thought: if we now trim the name for whitespace, would it make sense to do the same for the `$address`? Doing `trim($address)` in the parent class?

Reply:
From what I can make from the [RFC](https://tools.ietf.org/html/rfc5322#section-3.4) it looks like no.

But yes, should probably trim address as well. 

Is there any advantage to using preg_replace over str_replace? str_replace should be faster for these kind of operations, no?